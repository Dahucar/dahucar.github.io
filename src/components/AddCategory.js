import React, { useState } from 'react';
import PropTypes from 'prop-types'

// el metodo enviado desde otro componente se recive en los props
const AddCategory = ({ setCategories }) => {
  const [ inputValue, setInputValue ] = useState('');
  const handleChangeInputValue = ( e ) => {
    setInputValue( e.target.value );
  }

  const handleSubmitForm = ( e ) => {
    console.log('handleSubmitForm', inputValue);
    e.preventDefault();
    // a qui se llama al metodo que se recive desde los props en el por referencia se tendra el listado de categorias.

    if (inputValue.trim().length >= 2) {
      setCategories( categ =>  [ inputValue, ...categ ] );
      setInputValue( '' );
    }

  } 

  return (
    <form onSubmit={ handleSubmitForm }>
      <input 
        type="text"
        value={ inputValue }
        placeholder="Ingrese un nombre"
        onChange={ handleChangeInputValue }
      />
      <span>{ inputValue }</span>
    </form>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}

export default AddCategory;