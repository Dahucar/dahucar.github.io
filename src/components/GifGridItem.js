import React from 'react'

export const GifGridItem = ({ title, url }) => {
  return (
    <div className="card 
                    animate__animated 
                    animate__bounce
                    animate__fadeIn">
      <h4>{ title }</h4>
      <img src={ url } alt={ title } />
    </div>
  )
}

