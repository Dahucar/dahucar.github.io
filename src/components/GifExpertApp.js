import React, { useState } from 'react';
import AddCategory from './AddCategory';
import { GifGrid } from './GifGrid';

const GitExpertApp = () => {
  // el listado de categorias se enviara por refencia a los compoenentes que se deseen
  const [categories, setCategories] = useState([ 'Halo' ]);
  return (
    <>
      <h2>Gif Expert App</h2>
      {/* aqui se estan enviado por referencia las categorias y le metodo para guardar una nueva */}
      <AddCategory setCategories={ setCategories } />
      <hr />
      <ol>
        {
          categories.map( category => <GifGrid key={ category } category={ category } />)
        }
      </ol>
    </>
  )
}

export default GitExpertApp;