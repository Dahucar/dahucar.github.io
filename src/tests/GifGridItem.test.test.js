import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../components/GifGridItem';

describe('Testing on GifGridItem Component', () => {

  const url = 'https://test/url';
  const title = 'test title';
  const wrapper = shallow( <GifGridItem title={ title } url={ url } /> );    

  test('Should generate snapshot of GifGridItem Component', () => {
    expect( wrapper ).toMatchSnapshot();
  });

  test('Title and url properties should be assigned', () => {
    const image = wrapper.find( 'img' );
    expect( image.prop('src') ).toBe( url );
    expect( image.prop('alt') ).toBe( title );
  });

  test('The css class name should be assigned on the content div with the animate__fadeOut properti', () => {
    const classNameDiv = wrapper.find( 'div' ).prop('className'); 
    expect( classNameDiv.includes( 'animate__fadeOut' ) ).toBe( true );
  });
  
  
});
