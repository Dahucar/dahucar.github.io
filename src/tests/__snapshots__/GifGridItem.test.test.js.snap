// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on GifGridItem Component Should generate snapshot of GifGridItem Component 1`] = `
<div
  className="card animate__animated animate__bounce animate__fadeOut"
>
  <h4>
    test title
  </h4>
  <img
    alt="test title"
    src="https://test/url"
  />
</div>
`;
